### PR TITLE
Fix step invalidation

### PR DESCRIPTION
**Pull request by Rob0h**
What I Did
------------
- Fix step invalidation

How I Did it
------------
- Allow steps to be executed multiple times in order for steps to be invalidated properly

How to verify it
------------
- Tests
- Run `ship init` and then execute a step that invalidates others, invalidated steps will execute again to show new results

Description for the Changelog
------------
- Fix step invalidation

Picture of a Boat (not required but encouraged)
------------
🛶

### CODE REVIEW

File: pkg/lifecycle/daemon/routes_navcycle_completestep.go [hunk 1]
```diff
@@ -37,43 +36,25 @@ func (d *NavcycleRoutes) completeStep(c *gin.Context) {
...
-			}
-		}
-		_, stepAlreadyComplete := lifecycle.StepsCompleted[step.Shared().ID]
-
 		progress, ok := d.StepProgress.Load(step.Shared().ID)
-		shouldExecute := !ok || progress.Detail == `{"status":"success"}` && !stepAlreadyComplete
-
-		debug.Log("shouldExecute", shouldExecute)
+		shouldExecute := !ok || progress.Detail == `{"status":"success"}`
 		if shouldExecute {
```
Comment (laverya) [inline pkg/lifecycle/daemon/routes_navcycle_completestep.go]: What do you do if this is false?
Do we want to, you know, not execute the step?

Comment (Rob0h) [reply pkg/lifecycle/daemon/routes_navcycle_completestep.go]: no we're always going to "re-execute"

Comment (laverya) [reply pkg/lifecycle/daemon/routes_navcycle_completestep.go]: Because it looks like we execute either way

Comment (Rob0h) [reply pkg/lifecycle/daemon/routes_navcycle_completestep.go]: that's the intent. i'm removing to avoid confusion

**Pull request merged**